sqsjobs/item.go

```diff
@@ -161,36 +167,54 @@ func (i *Item) Nack() error {
 		return nil
 	}
 
-	// requeue message
-	err := i.Options.requeueFn(context.Background(), i)
-	if err != nil {
-		return err
-	}
+	if !i.Options.RetainFailedJobs {
+		// requeue as new message
+		err := i.Options.requeueFn(context.Background(), i)
+		if err != nil {
+			return err
+		}
+		// Delete original message
+		_, err = i.Options.client.DeleteMessage(context.Background(), &sqs.DeleteMessageInput{
+			QueueUrl:      i.Options.queue,
+			ReceiptHandle: i.Options.receiptHandler,
+		})
 
-	_, err = i.Options.client.DeleteMessage(context.Background(), &sqs.DeleteMessageInput{
-		QueueUrl:      i.Options.queue,
-		ReceiptHandle: i.Options.receiptHandler,
-	})
+		if err != nil {
+			return err
+		}
+	} else if i.Options.ErrorVisibilityTimeout > 0 {
```

Comment:
```go
	switch {
	case !i.Options.RetainFailedJobs:
	case i.Options.ErrorVisibilityTimeout > 0:
	default:
		return nil
	}
```
I think it'd be more readable.

Reply:
I changed it, but I think using an if-else is better because it makes it clear that the *first* switch case takes priority. I don't know how switch works in Go, but to me it seems safer to start with the first condition and then proceed to else, instead of trying to make it a jump-table (what if it jumps to the second case, if it matches, but first doesn't - we want to ignore `ErrorVisibilityTimeout` in all cases if `RetainFailedJobs` is false).

Reply:
It works as a jump table if you have a condition after switch, e.g. `switch foo`. In our case, we are just hoping for the first matching condition: https://go.dev/tour/flowcontrol/11
BTW, you can pass this short course to be more familiar with Go 😃

Reply:
"This construct can be a clean way to write long if-then-else chains." - that's all I need to know. Just wanted to make sure that 1 option was always evaluated first.